### DISCUSSION TITLE
Attributes stored in the database

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#957] [General]
Hello. It has come to my [attention](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/953) that this integration has a habit of storing a large amount of data in the database. I plan to try and reduce the attribute data that is stored by the recorder/database. This includes excluding the following

* data_last_retrieved
* mpan
* mprn
* serial_number
* is_export
* is_smart_meter
* target_times_last_evaluated

However, before I do, I wanted to know from people who use the data stored by the recorder to know what bits of information is important/used. 

Note: This will not effect the data available via the UI or the restoring of the above attributes upon restart.

**Comment by gcoan** [comment]
I think target_times_last_evaluated is useful to keep on a target rate sensor, but the rest and account id (on target rate sensor) are not of much use, or at least are not worth repeating in attributes lots of times

**Reply by BottlecapDave** [reply]
The target_times_last_evaluated would still be on the sensor and restored, it just wouldn't be stored in the database.

**Reply by gcoan** [reply]
> The target_times_last_evaluated would still be on the sensor and restored, it just wouldn't be stored in the database. 

works for me, no need to store it in the database as historical data but useful to be able to interrogate it
